cloudformation/jolly-roger.yaml

```diff
@@ -519,22 +518,80 @@ Resources:
+                    # server port the client connected to
+                    map $http_x_forwarded_port $proxy_x_forwarded_port {
+                      default $http_x_forwarded_port;
+                      ''      $server_port;
+                    }
+                    # Set appropriate X-Forwarded-Ssl header based on $proxy_x_forwarded_proto
+                    map $proxy_x_forwarded_proto $proxy_x_forwarded_ssl {
+                      default off;
+                      https on;
+                    }
+                    # If we receive Upgrade, set Connection to "upgrade"; otherwise, delete any
+                    # Connection header that may have been passed to this server
+                    map $http_upgrade $proxy_connection {
+                      default upgrade;
+                      '' close;
+                    }
+
+                    gzip_types text/plain text/css application/javascript application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
+
+                    log_format vhost '$host $remote_addr - $remote_user [$time_local] '
+                                    '"$request" $status $body_bytes_sent '
+                                    '"$http_referer" "$http_user_agent" '
+                                    '"$upstream_addr"';
+                    access_log off;
+
+                    # HTTP 1.1 support
+                    proxy_http_version 1.1;
+                    proxy_buffering off;
+                    proxy_set_header Host $http_host;
+                    proxy_set_header Upgrade $http_upgrade;
+                    proxy_set_header Connection $proxy_connection;
+                    proxy_set_header X-Real-IP $remote_addr;
+                    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+                    proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
+                    proxy_set_header X-Forwarded-Ssl $proxy_x_forwarded_ssl;
+                    proxy_set_header X-Forwarded-Port $proxy_x_forwarded_port;
+                    # Mitigate httpoxy attack
+                    proxy_set_header Proxy "";
                     proxy_cache_path /var/cache/nginx/cache levels=1:2 keys_zone=assets:10m;
-                - path: /etc/nginx-https/vhost.d/${AppUrl}
-                  content: |
-                    add_header Strict-Transport-Security "max-age=31536000";
                     proxy_cache assets;
-                    gzip on;
-                    gzip_comp_level 9;
-                    gzip_proxied any;
 
                     set_real_ip_from 10.0.0.0/8;
                     real_ip_header X-Forwarded-For;
 
-                    location /healthcheck {
-                      return 200 "OK\n";
+                    upstream jolly-roger {
+                      server 127.0.0.1:3000;
+                    }
+
+                    server {
+                      server_name deathandmayhem.com;
```

Comment:
do you mean `${AppUrl}` (or whatever the appropriate substitution would be) here?